### PR TITLE
resolved code formatting issue #371

### LINKED ISSUE #371: Code formatting issue in the release notes of Draft 2020-12
#### Describe the bug
When accessing to the release notes of Draft 2020-12:
https://json-schema.org/draft/2020-12/release-notes

There is a problem with missing formatting of the json code of the section "$dynamicRef and $dynamicAnchor": 
https://json-schema.org/draft/2020-12/release-notes#dollardynamicref-and-dollardynamicanchor

Screenshot 2024-02-27 at 00 47 02

#### How to Reproduce
Access the url: https://json-schema.org/draft/2020-12/release-notes#dollardynamicref-and-dollardynamicanchor

#### Expected behavior
Formatted code like other sections: 
Screenshot 2024-02-27 at 00 49 53

### PR DESCRIPTION
**Pull request by Adity20**
GitHub Issue: https://github.com/json-schema-org/website/issues/371

Summary: This pull request is being made proactively to fix a particular problem which is code formatting for the issue #371.
Resolves #371 

Screenshot 2024-02-28 181455
the UI looks like this now and if any other changes are required kindly let me @benjagm.

@benjagm kindly review my PR

### CODE REVIEW

File: pages/draft/2020-12/release-notes.md [hunk 1]
```diff
@@ -196,7 +200,7 @@ Here's how you would covert a schema using `$recursiveRef` to use `$dynamicRef`.
             }
           }
         }
-        // strict-tree schema, guards against misspelled properties
+      // strict-tree schema, guards against misspelled properties
```
Comment (gregsdennis) [inline pages/draft/2020-12/release-notes.md:203]: Looks like there's still an issue with syntax highlighting. Probably is this comment.

Can you change the language specifier to `jsonc` to see if that fixes the issue?

**Review by gregsdennis — CHANGES_REQUESTED** [review]

**Comment by Adity20** [comment]
Sure @gregsdennis I'll try that too.

**Comment by Adity20** [comment]
Hi @gregsdennis  I have tried some changes and now the UI looks like this 
Screenshot 2024-02-29 163342
Screenshot 2024-02-29 163351
kindly review it and let me know if some more changes are required

**Review by gregsdennis — CHANGES_REQUESTED** [review]

File: pages/draft/2020-12/release-notes.md [hunk 2]
```diff
@@ -196,7 +205,33 @@ Here's how you would covert a schema using `$recursiveRef` to use `$dynamicRef`.
...
+  </tr>
+  <tr>
+    <th colspan="2"> strict-tree schema, guards against misspelled properties</th>
+    <th colspan="2"> strict-tree schema, guards against misspelled properties</th>
+  </tr>
+  <tr>
+    <td>
+
+      ```json
+        {
```
[the remark this answers is not in the document]
Comment (Adity20) [reply pages/draft/2020-12/release-notes.md:220]: sure @gregsdennis i'll make the changes asap

[the remark this answers is not in the document]
Comment (Adity20) [reply pages/draft/2020-12/release-notes.md:208]: can you check now @gregsdennis

[the remark this answers is not in the document]
Comment (gregsdennis) [reply pages/draft/2020-12/release-notes.md:208]: Did you push the changes?  I don't see a difference in the code.  Also, could you provide a screenshot? I can't run the site locally.

[the remark this answers is not in the document]
Comment (Adity20) [reply pages/draft/2020-12/release-notes.md:208]: but at my end its showing the changes

[the remark this answers is not in the document]
Comment (Adity20) [reply pages/draft/2020-12/release-notes.md:208]: here are the screenshots 
Screenshot 2024-03-02 114910
Screenshot 2024-03-02 114924
Screenshot 2024-03-02 114939

and the output of these changes 
Screenshot 2024-02-29 163342
Screenshot 2024-02-29 163351

**Comment by Adity20** [comment]
If you are not able to see these new changes then I can close this pr and push the changes again by opening another pull request all changes at once if thats fine?

File: pages/draft/2020-12/release-notes.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gregsdennis) [reply pages/draft/2020-12/release-notes.md:208]: I don't see the indention changes I asked for.  This is what's confusing.

Please move the JSON all the way to the left, ignoring the HTML indentation.

[the remark this answers is not in the document]
Comment (Adity20) [reply pages/draft/2020-12/release-notes.md]: Now fine @gregsdennis ?
Screenshot 2024-03-02 140956

**Comment by Adity20** [comment]
Hi @gregsdennis I changed the format to jsonc and now the output looks like this kindly check 
Screenshot 2024-03-02 221450

File: pages/draft/2020-12/release-notes.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gregsdennis) [reply pages/draft/2020-12/release-notes.md]: There's still some indention.  Please pull _all the way_ to the left.

image

**Comment by gregsdennis** [comment]
What's going on here?

image

Also, the indentation seems to have been messed up on the 2020-12 and completely removed on the 2019-09.  It's almost as if it's rendering spaces with variable width.

image

This is very confusing; the code looks fine to me.

**Comment by Adity20** [comment]
Now fine @gregsdennis ?
Screenshot 2024-03-03 083545

**Comment by gregsdennis** [comment]
The code looks good, but it still doesn't render right.  The code has the correct indentation, but the render shows none.

**Comment by Adity20** [comment]
So what should I do now?

**Comment by benjagm** [comment]
Maybe this is related with the problems here:

There are 2 config attributes in the code editor I needed to change to avoid extra width of the containing page for some scenarios for small devices, but the solution was suboptimal. I think this is related. Below the commit I did:
https://github.com/json-schema-org/website/commit/cd3078604f28ecc1e9c3ad292e399b06cc97c17c

related (maybe): https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/376

**Comment by Adity20** [comment]
Is this for issue #393?

**Comment by Adity20** [comment]
> Maybe this is related with the problems here:
> 
> There are 2 config attributes in the code editor I needed to change to avoid extra width of the containing page for some scenarios for small devices, but the solution was suboptimal. I think this is related. Below the commit I did: [cd30786](https://github.com/json-schema-org/website/commit/cd3078604f28ecc1e9c3ad292e399b06cc97c17c)
> 
> related (maybe): [react-syntax-highlighter/react-syntax-highlighter#376](https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/376)

Is this for issue #393 @benjagm ?

**Comment by Adity20** [comment]
> Maybe this is related with the problems here:
> 
> There are 2 config attributes in the code editor I needed to change to avoid extra width of the containing page for some scenarios for small devices, but the solution was suboptimal. I think this is related. Below the commit I did: [cd30786](https://github.com/json-schema-org/website/commit/cd3078604f28ecc1e9c3ad292e399b06cc97c17c)
> 
> related (maybe): [react-syntax-highlighter/react-syntax-highlighter#376](https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/376)

So what new changes I need to introduce @benjagm ?

**Comment by benjagm** [comment]
The problem is with indentation right? Those 2 parameters affects exactly that. First play with them, later well need to check how to make sure the  component's behavior is correct in other pages, specifically in small devices.

**Comment by Adity20** [comment]
Okay @benjagm I'll try that too.

**Comment by benjagm** [comment]
@Adity20 did you tested the component indentation disabling the attributes wrapLines and wrapLongLines? I think that can fix it.

**Comment by Adity20** [comment]
I tried disabling these two attributes and tried some other changes too but that's creating an overflow in the page as the code is taking more space now to be in the same line. 
I don't think this will work we'll have to apply more custom styles maybe

**Comment by benjagm** [comment]
> Can we have each section scroll horizontally? Would it be weird to have that here but not in other places?
> 
> Alternatively, what if we made it look like a git diff, where you do a red `-` line followed by a green `+`line? Is that possible?

I'd suggest fix everything but the indentation issue here and create an different issue to configure the code component to have its own horizontal scroll bar se fix this across the entire site.

**Comment by gregsdennis** [comment]
I'm happy with that

**Comment by Adity20** [comment]
So I should try horizontal scrollbar once or keep trying some other changes?

**Comment by benjagm** [comment]
@Adity20 I think si better to do it in a separate issue. Let suggest this:

- Let's approve this leaving the indentation unfixed.
- I'll open the issue and assign it to you if this is ok for you.

**Review by benjagm — APPROVED** [review]
Great job @Adity20 !!!

**Comment by Adity20** [comment]
Yeah that works for me @benjagm. I can start working on the that one then.

**Comment by benjagm** [comment]
@Adity20 please add a comment in https://github.com/json-schema-org/website/issues/456 to get assigned.

**Review by gregsdennis — APPROVED** [review]
Everything except the indentation seems to be resolved.  I'm happy address that elsewhere.

**Pull request merged**